### ISSUE TITLE
Treeview's svg control click target is small

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#6670]
related #6668 

The click target size for the SVG button control in Treeview is quite small potentially negatively impacting the [following user types](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html):
- Users who use a mobile device where touch screen is the primary mode of interaction;
- Users with mobility impairments such as hand tremors;
- Users who use a mobile device in environments where they are exposed to shaking such as public transportation;
- Users who have difficulty with fine motor movements;
- Users who access a device using one hand;
- Users with large fingers, or who are operating the device with only a part of their finger or knuckle;
- Users who have low vision may better see the target.

Currently in WCAG 2.1 a 44px by 44px target size is a AAA recommendation so the control in question is not a violation at time of writing. That said [WCAG 2.2](https://www.w3.org/TR/WCAG22/) (in Working Draft status) is looking to elevate this recommendation to AA level which makes it a concern. I think it's worth discussing what/how a larger click target could be used here.

**Comment by dakahn** [comment]
@designertyler 🏄

**Comment by carmacleod** [comment]
From @dakahn in https://github.com/carbon-design-system/carbon/pull/6008#issuecomment-673503210:

> have the list item itself be clickable. ... give users a larger click target make the tree more usable.

I agree that we need to do something to increase the click target of expandable treeitems.

I understand that Carbon wants to keep the "don't expand on click" behavior, so here's a suggestion: the Windows Explorer tree expands when you double-click on an expandable treeitem (the Mac Finder kind of does something similar, although double-click "expands" by showing the contents of the folder in a reduced tree). So consider adding "double-click expands" behavior. You might want to make it a prop? i.e. doubleClickExpands or singleClickExpands - depending on which one would be a better default behavior for Carbon use cases. Note that single-click expansion is pretty common for trees, too. Check out these trees:

- https://w3c.github.io/aria-practices/examples/treeview/treeview-1/treeview-1a.html
- https://dequeuniversity.com/library/aria/tree-view
- https://material-ui.com/components/tree-view/

**Comment by dakahn** [comment]
closing in favor or #6792
